app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js

```diff
@@ -269,13 +270,30 @@ export default class AdminSidebarPanel extends BaseCustomSidebarPanel {
     const navMap = savedConfig || ADMIN_NAV_MAP;
 
     if (!session.get("safe_mode")) {
-      navMap.findBy("name", "plugins").links.push(...pluginAdminRouteLinks());
+      const pluginLinks = navMap.findBy("name", "plugins").links;
+      pluginAdminRouteLinks().forEach((pluginLink) => {
+        if (!pluginLinks.mapBy("name").includes(pluginLink.name)) {
+          pluginLinks.push(pluginLink);
+        }
+      });
+
       this.adminSidebarStateManager.setLinkKeywords(
         "admin_installed_plugins",
         installedPluginsLinkKeywords()
       );
     }
 
+    store.findAll("theme").then((themes) => {
```

Comment:
I do worry about having an AJAX request here; this is why I added `visiblePlugins` to the preload store for admins:

https://github.com/discourse/discourse/blob/e1d9fd479f83191af24a10964a0e9915de6141f5/app/controllers/application_controller.rb#L674-L688

I think it might be best if we store a simplified array of themes & components preloaded for admin for this purpose, e.g. something like:

```ruby
Theme.include_relations.order(:name).to_a.map { |t| [t.name, t.component, t.enabled] }
```

You are right that eventually we want some serialized search data index for admin with all routes and keywords for the admin-wide search down the road -- I think also we would preload this, then we would be able to get rid of `visiblePlugins` and whatever this above preload store is called (`installedThemes`?)

Reply:
We talked about this -- it's fine for now especially since it's just adding keywords and not blocking adding links. When we do the proper admin search we need to rethink all this search data anwyay.